src/particles/ImpactXParticleContainer.cpp

```diff
@@ -181,25 +208,27 @@ namespace impactx
         amrex::ParticleReal const * const AMREX_RESTRICT py_ptr = py.data();
         amrex::ParticleReal const * const AMREX_RESTRICT pt_ptr = pt.data();
 
-        amrex::ParallelFor(np,
-        [=] AMREX_GPU_DEVICE (int i) noexcept
+        amrex::ParallelFor(num_to_add,
+            [=] AMREX_GPU_DEVICE (int i) noexcept
         {
-            idcpu_arr[old_np+i] = amrex::SetParticleIDandCPU(pid + i, cpuid);
+            idcpu_arr[old_np+i] = amrex::SetParticleIDandCPU(pid + my_index + i, cpuid);
 
-            x_arr[old_np+i] = x_ptr[i];
-            y_arr[old_np+i] = y_ptr[i];
-            t_arr[old_np+i] = t_ptr[i];
+            x_arr[old_np+i] = x_ptr[my_index+i];
```

Comment:
Does `my_index` also need to be added to the index on the LHS? e.g.,
```suggestion
            x_arr[old_np+my_index+i] = x_ptr[my_index+i];
```